api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.pmiops.workbench.survey;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.UserAccessTierDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserAccessTier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NewUserSatisfactionSurveyServiceImpl implements NewUserSatisfactionSurveyService {
+
+  private final UserAccessTierDao userAccessTierDao;
+  private AccessTierService accessTierService;
+  private final Clock clock;
+
+  @Autowired
+  public NewUserSatisfactionSurveyServiceImpl(
+      UserAccessTierDao userAccessTierDao, AccessTierService accessTierService, Clock clock) {
+    this.userAccessTierDao = userAccessTierDao;
+    this.accessTierService = accessTierService;
+    this.clock = clock;
+  }
+
+  @Override
+  public boolean eligibleToTakeSurvey(DbUser user) {
+    if (user.getNewUserSatisfactionSurvey() != null) {
+      return false;
+    }
+
+    DbAccessTier registeredAccessTier = this.accessTierService.getRegisteredTierOrThrow();
+
+    Optional<DbUserAccessTier> userRegisteredAccessTier =
+        this.userAccessTierDao.getByUserAndAccessTier(user, registeredAccessTier);
+
+    if (!userRegisteredAccessTier.isPresent()) {
+      return false;
+    }
+
+    Instant enabledTime = userRegisteredAccessTier.get().getFirstEnabled().toInstant();
+
+    final Instant twoWeeksAgo = clock.instant().minus(2 * 7, ChronoUnit.DAYS);
+    final boolean enabledMoreThanTwoWeeksAgo = enabledTime.isBefore(twoWeeksAgo);
+
+    final Instant twoMonthsTwoWeeksAgo = twoWeeksAgo.minus(61, ChronoUnit.DAYS);
+    final boolean eligibleForLessThanTwoMonths = enabledTime.isAfter(twoMonthsTwoWeeksAgo);
```

Comment:
I like these clear variable names.  But I'm not sure if this one is quite right.  Could you add some comments describing the precise window we're aiming for here?

Reply:
I added a comment, but I'm not sure if it's better. Here's the window we're looking for:

- Two weeks after RT access, users can take the survey
- Two months after ^this^, users become ineligible to take the survey

Let me know if you have a better idea.

Reply:
hmmmm.  How about:

windowStart = twoWeeksAndTwoMonthsAgo;
windowEnd = twoWeeksAgo;

return (firstEligibleDate > windowStart) && (firstEligibleDate < windowEnd);

Reply:
Ooh thanks for the nudge, I [inverted the logic](https://github.com/all-of-us/workbench/pull/7135/commits/0c0a90ff52638dbe9e5e9deac4c91d716942f19a) and it's a LOT clearer now.